packages/hackmcx-api/src/controllers/users.controller.js

```diff
@@ -21,3 +21,46 @@ export async function getUsersByUserId(req, res){
             }
         })
 }
+
+export async function postUser(req, res){
+    const emptyUsername = !req.body.username || /^\s*$/.test(req.body.username);
+    const emptyFirstName = !req.body.firstname || /^\s*$/.test(req.body.firstname);
+    const emptyLastName = !req.body.lastname || /^\s*$/.test(req.body.lastname);
+    const usernameExists = (await dbClient.select('username').from('users').where('username', req.body.username)).length > 0 ? true : false;
```

Comment:
We might be able to look for an exception on insert instead of doing an explicit check here just to reduce the total number of DB calls.